Controller/src/main/java/org/apidb/irods/ProcessIrodsEvents.java

```diff
@@ -29,23 +29,72 @@
  * intended for iRODS, it has no specific references to iRODS.  It could be used
  * with a POSIX system as well.
  *
+ * <h2>Runtime Configuration</h2>
+ *
+ * This script requires the following configuration to be set on the environment
+ * to execute:
+ *
+ * <dl>
+ *   <dt><code>PROJECT_ID</code></dt>
+ *   <dd><b>REQUIRED</b> Target project that events should be processed for.
+ *       Events that are for other projects will be skipped.</dd>
+ *   <dt><code>DATASET_STORE_ID</code></dt>
+ *   <dd><b>REQUIRED</b> Identifies the datastore associated with the events to
+ *       be handled.  This value is used to ensure that this datastore matches
+ *       the one the current build supports.</dd>
+ *   <dt><code>RUN_MODE</code></dt>
+ *   <dd><b>REQUIRED</b> Selects the operation mode the script will execute in.
+ *       Must be one of &quot;<code>sync</code>&quot; or
+ *       &quot;<code>cleanup</code>&quot;.  <code>sync</code> mode synchronizes
+ *       the datasets from iRODS into the database.  <code>cleanup</code> mode
+ *       performs cleanup steps to resolve issues encountered when last syncing
+ *       datasets.</dd>
+ *   <dt><code>MAX_EVENTS</code></dt>
+ *   <dd><b>OPTIONAL</b> Sets the maximum number of events to process in a
+ *       single execution of the script.  This value paginates the events coming
+ *       from iRODS to avoid issues with connection leaks that are encountered
+ *       on long running Jargon-iRODS sessions.<br><br>Defaults to 150</dd>
+ * </dl>
+ *
  * @author crisl-adm
  */
 public class ProcessIrodsEvents {
 
+  // region Environment configuration variables
+
+  private static final String ENV_PROJECT_ID = "PROJECT_ID";
+
+  private static final String ENV_DATASTORE_ID = "DATASET_STORE_ID";
+
+  private static final String ENV_RUN_MODE = "RUN_MODE";
+
+  private static final String ENV_MAX_EVENTS = "MAX_EVENTS";
+
+  // endregion Environment configuration variables
+
+
   private static final Logger logger = Logger.getLogger(ProcessIrodsEvents.class);
+
+  /**
+   * Default maximum number of events to handle per script execution.
+   *
+   * This value may be overridden using the environment variable
+   * {@code MAX_EVENTS}.
+   */
+  private static final int DEFAULT_MAX_EVENTS_PER_EXEC = 150;
+
   public static final String EVENTS_DIR = "/ebrc/workspaces/events";
 
   public static void main(String[] args) throws Exception {
 
-	// The id of the project for which these events are intended.
-    String projectId = System.getenv("PROJECT_ID");
+	  // The id of the project for which these events are intended.
+    String projectId = System.getenv(ENV_PROJECT_ID);
```

Comment:
Consider using [this](https://github.com/VEuPathDB/FgpUtil/blob/master/Core/src/main/java/org/gusdb/fgputil/runtime/Environment.java).  Should be available.